packages/dds/tree/docs/undo/README.md

```diff
@@ -169,6 +169,95 @@ Partial undo would allow such a user to only undo the changes within the region
 This may need to be characterized as a combination of input context regions and output context regions.
 
+## V1 Undo
+
+Here we detail the vision for the first implementation of undo.
+
+This first version aims to achieve some basic undo functionality with a minimum amount of code changes and complexity.
+To that end, we mostly reuse the existing code paths for changesets by always sending concrete undos over the wire.
+The undo edit is created by inverting the edit that needs to be undone,
+and rebasing that inverse over all the changes that have been applied since.
+
+Sending concrete undo edits alleviates the need establish and maintain distributed consensus on an undo window.
+It does however require sending rebased changes over the wire.
+
+### Concrete Undo of Unsequenced Changes
+
+Using concrete undos even when the change to be undone has not been sequence is somewhat problematic
+because we cannot know in advance the exact impact of the change to undo.
+
+For this V1, we simply produce a "best attempt" undo based on the most up to date version of the change to be undone.
+This could lead to some data loss in scenarios where the change to be undone deletes a subtree under which content is concurrently inserted.
+
+### Undo Tree
+
+In order to perform an undo operation, it is necessary that we are able to determine which prior edit is to be undone.
+To that end, we need to maintain a tree of undoable commits.
+That tree is a sparse copy of the commit tree maintained by the `EditManager` for branch management.
+
+The structure forms a tree as opposed to a linked-list because different local branches can share the same ancestor commits.
+Each branch however only ever sees a single spine of this tree, which therefore looks like a linked-list to said branch.
+
+The tree is sparse because it does **_not_** contain the following kinds of edits:
+
+-   Edits authored by other clients (this is only a concern for the part of the tree that represents the trunk).
+-   Edit authored on other branches.
+-   Undo edits.
+
+Note that some of these edits in the tree may be part of the trunk while others may be on a branch.
+Each branch need only maintain a "head" pointer to the child-most commit on the branch.
+
+#### Adding New Commits
+
+When a new edit is made by the local client, one of two things must happen:
+
+-   If the edit is not an undo then an undoable commit node is added under the head undoable commit maintained by the branch on which the edit was made.
+    This new undoable commit becomes the new head undoable commit for this branch.
+-   If the edit is an undo then the head undoable commit maintained by the branch on which the edit was made is popped.
+    The parent of the undone commit becomes the new head undoable commit for this branch.
+
+#### Forking
+
+When a branch is forked
+(whether from the trunk or an existing branch)
```

Comment:
I see. I had not realized that in our current implementation it's the local branch that is special as opposed to the trunk. The parenthetical should be removed here.

Reply:
Right. In my mind the local branch is special because:
1. It dictates the view of the ISharedTree (the root, default Checkout)
2. It automatically pulls whenever a sequenced edit comes in
3. It automatically merges whenever its own edit gets sequenced

I think point number 1 above is the main reason it makes sense to have branches come off of the local branch rather than directly off of the trunk. Once your branch gets "merged all the way up", I think you want to see its changes reflected in your view optimistically before sequencing, i.e. be in the local branch.